### DISCUSSION TITLE
MapLayers Needed to Replicate iTwin Viewer 2.7.0

### DISCUSSION THREAD

**Discussion by mattxerini** [iTwin/itwinjs-core#5513] [Q&A] [answered]
Hi, 

I am trying to update a project I am working on with the newest iTwin 3.7.0. 
As per documention I understand the API keys of several third-party applications have been removed from iTwin and now it is up to the user to create and provide those keys.

So I already created a key for Cesium Terrain following the documentations, however now I am not sure what kind of Map Layers was used under the hood by iTwin and therefore if I need other keys.

This is the viewer version 2.7.0, can someone explain what other API key I would need to replicate the below viewer?
<img width="1225" alt="image" src="https://github.com/iTwin/itwinjs-core/assets/123177586/485b7191-751a-4288-a265-cfae50c304cc">
<img width="1224" alt="image" src="https://github.com/iTwin/itwinjs-core/assets/123177586/5ef7349d-38db-4345-8ec6-567d86ff768b">
<img width="1224" alt="image" src="https://github.com/iTwin/itwinjs-core/assets/123177586/6f2fcdf1-e71e-4e8f-8cd1-242a91b6e0c3">

**Comment by pratikshan85** [comment]
@mattxerini We are looking in to the issue.

**Comment by pmconne** [comment]
Hi @mattxerini. I assume you already saw the [3.0 changelog](https://github.com/iTwin/itwinjs-core/blob/master/docs/changehistory/3.0.0.md#removed-default-api-keys) describing API key removals. The image in your screenshot is almost certainly using Bing Maps.

**Reply by mattxerini** [reply]
Yes, as I mentioend above, I saw the changelog and created a Cesium account and the access token. 
Although from the previous code implemented I cannot understand what other keys I should use, as I assume the layers were used already under the hood by the old ITwin.

**Reply by mattxerini** [reply]
Could someone confirm what layers I would need and if so can I use the Cesium token or I need to create accounts for each of the layers? i.e BingMap, MapBox etc.

**Reply by pratikshan85** [reply]
@mattxerini Please check the below link :

https://github.com/iTwin/itwinjs-core/blob/master/docs/changehistory/3.0.0.md#removed-default-api-keys

**Reply by mattxerini** [reply]
Again as I said above, I saw that and created A valid [Cesium ION](https://cesium.com/platform/cesium-ion/) key.

Now the questions is what other keys were used under the hood. The old code is not setting any provider with value BingMap or MapBox. However right now the viewer is black and therefore I am asking looking at the viewer older version what MapLayers were used by default by iTwin

**Reply by pmconne** [reply]
It defaults to Bing.

**Reply by mattxerini** [reply]
Ok, so I will have to pass a valid Bing key correct? Or can I use the Cesium token and add a Bing Arial layer asset into the Cesium account?

for example have the following assets in Cesium and use the Cesium Token for BingMap?
<img width="914" alt="image" src="https://github.com/iTwin/itwinjs-core/assets/123177586/b83faa44-0625-45bd-82f8-ebff969332d9">
****

**Reply by pmconne** [reply, marked as answer]
If you want to see Bing map imagery in an iTwin.js viewport, you must supply a Bing API key.
If you have questions about whether your Cesium account provides a Bing API key, please direct them to the folks at Cesium.

**Reply by mattxerini** [reply]
Thank you
